### DISCUSSION TITLE
Show only selected branches feature

### DISCUSSION THREAD

**Discussion by ddimkovikj** [gitextensions/gitextensions#9495] [Ideas]
Starting this discussion for a new feature I proposed in #9472, but something similar was already developed by @RussKie in #9412.
Ctrl + click was proposed as an implementation.
I was thinking of the following:
- Select desired branches
- Right click -> Show only (these)selected branches

This is in case multi-select branch context menu is supported in GE. From what I could see, it shows the same menu whether one or multiple branches are selected.

@RussKie let me know what you think.

**Comment by ddimkovikj** [comment]
Furthermore, it should automatically fill in the branch filter field in the toolbar with the selected branches, so you can reset it and view all branches again.

**Comment by h0lg** [comment]
I would love a more comfortable multi-branch filter as well.

For me however, the **natural interface for selecting multiple branches** to show would be the **left branch tree panel** instead of the revision grid. That's because selecting the branches (the heads of which may be far apart) from the revision grid may require a lot of scrolling. Edit: this has already been proposed in https://github.com/gitextensions/gitextensions/issues/3114#issuecomment-503377888 .

Additionally, the context menu of a branch in that panel already has an entry for _Show this branch only_ setting the branch filter of the revision grid - so much of the required functionality seems to be there already. Only thing that's missing is the ability to select multiple branches from the branch tree.

At that point, I wouldn't even need the branch filter in the top toolbar any more, rendering related feature requests obsolete for me: #9472 #491 #3546 #9412 #3165

**Comment by RussKie** [comment]
> * Right click -> Show only (these)selected branches

I think this can be useful.

There are multiple filter-related discussions and requests (@h0lg was kind enough to trawl through and link most of them above) and all of them need to be considered collectively. E.g. 
- the quick filter should be easily accessible, 
- the quick and the advanced filters must be in sync, 
- filtering must be possible from all parts of the app (i.e. the left panel, the revision grid, the toolbar, etc.),
- it should be possible to add/remove/reset the filter from context menus, etc.

This goes into the mix as well: https://github.com/gitextensions/gitextensions/issues/3546#issuecomment-964052931

I've spent some time in the current dev cycle to address or make progress in some of these points, and the filtering should be more approachable in the next release. But I don't have bandwidth to complete this epic, nor it is a priority for me personally.

**Comment by h0lg** [comment]
This is taken care of by PR #9723.
